x/dex/keeper/core.go

```diff
@@ -182,27 +182,33 @@ func (k Keeper) WithdrawCore(goCtx context.Context, msg *types.MsgWithdrawl, tok
 	}
 	totalReserve0ToRemove := sdk.ZeroInt()
 	totalReserve1ToRemove := sdk.ZeroInt()
+	feeTiers := k.GetAllFeeTier(ctx)
 
 	for i, feeIndex := range msg.FeeIndexes {
 		sharesToRemove := msg.SharesToRemove[i]
 		tickIndex := msg.TickIndexes[i]
 
-		feeValue, found := k.GetFeeTier(ctx, feeIndex)
-		if !found {
-			return types.ErrValidFeeIndexNotFound
+		// check that feeIndex is a valid index of the fee tier
+		if feeIndex >= uint64(len(feeTiers)) {
+			return sdkerrors.Wrapf(types.ErrValidFeeIndexNotFound, "(%d) does not correspond to a valid fee", feeIndex)
```

Comment:
Need to think more about how throwing errors inside the loops is going to work. Right now it is possible to burn shares if the first withdrawal succeeds but then we return early if a later deposit fails and we never finish the for loop and get to the SendCoins section

Reply:
it is my understanding that if an error is returned then the whole transaction should revert.